src/net_processing.cpp

```diff
@@ -1722,11 +1730,11 @@ static CTransactionRef FindTxForGetData(const CTxMemPool& mempool, const CNode&
     return {};
 }
 
-void static ProcessGetData(CNode& pfrom, const CChainParams& chainparams, CConnman& connman, CTxMemPool& mempool, const std::atomic<bool>& interruptMsgProc) LOCKS_EXCLUDED(cs_main)
+void static ProcessGetData(CNode& pfrom, PeerRef peer, const CChainParams& chainparams, CConnman& connman, CTxMemPool& mempool, const std::atomic<bool>& interruptMsgProc) EXCLUSIVE_LOCKS_REQUIRED(!cs_main, peer->cs_vRecvGetData)
```

Comment:
`PeerRef` is a shared pointer, so by passing it by value here, you're incrementing the ref count, and decrementing it again when you leave the function, which incurs a performance cost, since those operations need to be atomic.

This function doesn't need to take shared ownership of the underlying `Peer` object, so you should pass by `Peer&` (see http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r30-take-smart-pointers-as-parameters-only-to-explicitly-express-lifetime-semantics).

Reply:
thanks for the pointer! fixed.